### PR TITLE
refactor: Port MParticleWebView to Swift

### PR DESCRIPTION
**Pull request by einsteinx2**
## Summary
 - Port MParticleWebView to Swift
 - Refactor MPLogger.swift

 ## Testing Plan
 - [x] Was this tested locally? If not, explain why.
 - Smoke tested and confirmed working in test app, and all unit tests pass

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-6987

### CODE REVIEW

File: mParticle-Apple-SDK.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -527,6 +529,7 @@
 		531BCF322B28A23400F5C573 /* MPIdentityCaching.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = MPIdentityCaching.h; sourceTree = "<group>"; };
 		531BCF332B28A23400F5C573 /* MPIdentityCaching.m */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.objc; path = MPIdentityCaching.m; sourceTree = "<group>"; };
 		531BCF392B28A83E00F5C573 /* MPIdentityCachingTests.m */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.objc; path = MPIdentityCachingTests.m; sourceTree = "<group>"; };
+		534C11B22D08A73100466F71 /* MParticleWebView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = MParticleWebView.swift; sourceTree = "<group>"; };
```
[the remark this answers is not in the document]
Comment (einsteinx2) [reply mParticle-Apple-SDK.xcodeproj/project.pbxproj:528]: Whoops forgot to remove them from the project and delete the files. Just pushed up another PR.

File: mParticle-Apple-SDK/Utils/MParticleWebView.swift [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        super.init()
+    }
+    
+    @objc public func start(customUserAgent: String?, shouldCollect: Bool, defaultUserAgentOverride: String?) {
+        self.initializedDate = Date()
+        self.customUserAgent = customUserAgent
+#if os(iOS)
+        self.shouldCollect = shouldCollect
+#endif
+        self.defaultUserAgent = defaultUserAgentOverride ?? originalDefaultUserAgent
```
[the remark this answers is not in the document]
Comment (einsteinx2) [reply mParticle-Apple-SDK/Utils/MParticleWebView.swift:41]: It could be, but then we would need another private property to store the `defaultUserAgentOverride` value, so not sure it's worth doing.

File: UnitTests/MParticleWebViewTests.m [hunk 3]
```diff
@@ -48,84 +36,59 @@ - (void)testInit {
...
     _webView.initializedDate = [NSDate dateWithTimeIntervalSinceNow:-4];
     XCTAssertTrue([_webView shouldDelayUpload:5]);
 }
 
 - (void)testOriginalDefaultAgent {
     NSString *defaultAgent = [NSString stringWithFormat:@"mParticle Apple SDK/%@", MParticle.sharedInstance.version];
-    XCTAssertEqualObjects(_webView.originalDefaultAgent, defaultAgent);
-}
-
-- (void)testBackgroundCollection {
```
[the remark this answers is not in the document]
Comment (einsteinx2) [reply UnitTests/MParticleWebViewTests.m:117]: Yesh, so the 2 tests I removed were because OCMock wasn't properly detecting the call of the method (the "expect") even though it was 100% being called, confirmed via breakpoints. Seems to be some incompatibility with OCMock and Swift. In any case, when I looked closer at what was really being tested, it was just testing that a method was called that is literally always called, it wasn't really testing the logic.

Basically on iOS `evaluateAgent` is always called after calling `startWithCustomUserAgent` and on tvOS it never is, so this and the other test I removed ultimately were just testing that the `#if os(iOS)` worked, which isn't really useful.

**Review by BrandonStalnaker — APPROVED** [review]
LGTM
